src/components/Autocomplete.tsx

```diff
@@ -0,0 +1,107 @@
+import classNames from 'classnames';
+import { Person } from '../types/Person';
+import { useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
+import './Autocomplete.scss';
+
+type Props = {
+  people: Person[];
+  onSelected: (person: Person | null) => void;
+  delay?: number;
+};
+
+export const Autocomplete: React.FC<Props> = ({
+  people,
+  onSelected = () => {},
+  delay = 300,
+}) => {
+  const [focusOnInput, setFocusOnInput] = useState(false);
+  const [appliedQuery, setAppliedQuery] = useState('');
+  const [query, setQuery] = useState('');
+
+  const applyQuery = debounce(setAppliedQuery, delay);
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value;
+
+    setQuery(value);
+    applyQuery(value);
+    setFocusOnInput(true);
+    onSelected(null);
+  };
+
+  const handleSelectPerson = (person: Person) => {
+    onSelected(person);
+    setFocusOnInput(false);
+    setQuery(person.name);
+  };
+
+  const filteredPeople = useMemo(() => {
+    return people.filter(person =>
+      person.name
+        .toLowerCase()
+        .trim()
+        .includes(appliedQuery.toLowerCase().trim()),
+    );
+  }, [appliedQuery, people]);
+
+  return (
+    <div
+      className={classNames('dropdown', {
+        'is-active': focusOnInput,
+      })}
+    >
+      <div className="dropdown-trigger">
+        <input
+          type="text"
+          placeholder="Enter a part of the name"
+          className="input"
+          value={query}
+          data-cy="search-input"
+          onFocus={() => setFocusOnInput(true)}
+          onBlur={() => setFocusOnInput(false)}
+          onChange={handleInputChange}
+        />
+      </div>
+
+      {focusOnInput && (
+        <div className="dropdown-menu" role="menu" data-cy="suggestions-list">
+          <div className="dropdown-content">
+            {filteredPeople.length > 0 ? (
+              filteredPeople.map(person => (
+                <div
+                  key={person.slug}
```

Comment:
You are generating a key on render. This can lead to performance issues. It's recommended to generate keys during the creation of items. Here is a [link](https://medium.com/blackrock-engineering/5-common-mistakes-with-keys-in-react-b86e82020052) explaining why.